mods/xpanes/init.lua

```diff
@@ -1,156 +1,146 @@
-		if need[1] == true and need[3] == true then
-			need[1] = nil
-			need[3] = nil
-			table.insert(take, full_boxes[1])
-			table.insert(take2, sb_full_boxes[1])
+	local any = node.param2
+	local c = {}
+	local count = 0
+	for dir = 0, 3 do
+		c[dir] = connects_dir(pos, name, dir)
+		if c[dir] then
+			any = dir
+			count = count + 1
 		end
-		if need[2] == true and need[4] == true then
-			need[2] = nil
-			need[4] = nil
-			table.insert(take, full_boxes[2])
-			table.insert(take2, sb_full_boxes[2])
-		end
-		for k in pairs(need) do
-			table.insert(take, half_boxes[k])
-			table.insert(take2, sb_half_boxes[k])
-		end
-		local texture = def.textures[1]
-		if cnt == 1 then
-			texture = def.textures[1].."^"..def.textures[2]
+	end
+
+	if count == 0 then
+		swap(pos, node, name .. "_flat", any)
+	elseif count == 1 then
+		swap(pos, node, name .. "_flat", (any + 1) % 4)
+	elseif count == 2 then
+		if (c[0] and c[2]) or (c[1] and c[3]) then
+			swap(pos, node, name .. "_flat", (any + 1) % 4)
+		else
+			swap(pos, node, name, 0)
 		end
-		minetest.register_node(":xpanes:"..name.."_"..i, {
-			drawtype = "nodebox",
-			tiles = {def.textures[3], def.textures[3], texture},
-			paramtype = "light",
-			groups = def.groups,
-			drop = "xpanes:"..name,
-			sounds = def.sounds,
-			node_box = {
-				type = "fixed",
-				fixed = take
-			},
-			selection_box = {
-				type = "fixed",
-				fixed = take2
-			}
-		})
+	else
+		swap(pos, node, name, 0)
+	end
+end
+
+minetest.register_on_placenode(function(pos, node)
+	if minetest.get_item_group(node, "pane") then
+		update_pane(pos)
+	end
+	for i = 0, 3 do
+		local dir = minetest.facedir_to_dir(i)
+		update_pane(vector.add(pos, dir))
 	end
+end)
 
-	for k, v in pairs(pane_def_fields) do
-		def[k] = def[k] or v
+minetest.register_on_dignode(function(pos)
+	for i = 0, 3 do
+		local dir = minetest.facedir_to_dir(i)
+		update_pane(vector.add(pos, dir))
 	end
+end)
 
-	def.on_construct = function(pos)
-		update_pane(pos, name)
+xpanes = {}
+function xpanes.register_pane(name, def)
+	for i = 1, 15 do
+		minetest.register_alias("xpanes:" .. name .. "_" .. i, "xpanes:" .. name .. "_flat")
 	end
 
-	minetest.register_node(":xpanes:"..name, def)
+	local groups = table.copy(def.groups)
+	groups.pane = 1
+	minetest.register_node(":xpanes:" .. name .. "_flat", {
+		description = def.description,
+		drawtype = "nodebox",
+		paramtype = "light",
+		is_ground_content = false,
+		sunlight_propagates = true,
+		inventory_image = def.inventory_image,
+		wield_image = def.wield_image,
+		paramtype2 = "facedir",
+		tiles = {def.textures[3], def.textures[3], def.textures[1]},
+		groups = groups,
+		drop = "xpanes:" .. name .. "_flat",
+		sounds = def.sounds,
+		node_box = {
+			type = "fixed",
+			fixed = {{-1/2, -1/2, -1/32, 1/2, 1/2, 1/32}},
+		},
+		selection_box = {
+			type = "fixed",
+			fixed = {{-1/2, -1/2, -1/32, 1/2, 1/2, 1/32}},
+		},
+		connect_sides = { "left", "right" },
+	})
+
+	local groups = table.copy(def.groups)
```

Comment:
> Oh FCOL already.

Did not intend to frustrate you with it.
Rather to prevent the following PR-failings.

```lua
ignore = {"411"} -- `411  │ Redefining a local variable. `
```
Is still an option, as pointed out.